blood-sucker-tgui - Adds a new tgui interface for the bloodsucker antag

raffclar opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds a Bloodsucker TGUI interface by extending tgui with a new "fulpui" package.

## Why It's Good For The Game

Improved UX for the bloodsucker introduction window. Lays the groundwork for future fulpstation interfaces.

Existing Bloodsucker tips interface:
![image](https://user-images.githubusercontent.com/32263167/164351004-1f5c9780-116c-4901-953d-2be7690077e7.png)
New Bloodsucker tgui interface:
![image](https://user-images.githubusercontent.com/32263167/164351029-2d7060f2-4188-4dbc-bfdf-de533cb86b68.png)


## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Bloodsucker antag tgui interface
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

JohnFulpWillard commented on tgui/yarn.lock:
```diff
@@ -4320,6 +4320,24 @@ __metadata:
   languageName: node
   linkType: hard
 
+"fulpui-patches@workspace:packages/fulpui-patches":
```
this being in the tgui folder doesnt bother me, but it still feels like there's a ton of TG edits here. Is this part able to be a separate file or is that not how it works? I'm still very new to TGUI

raffclar replied:
This needs to be kept. The tgui package is dependent on it. I tried to do very some hacky stuff within webpack and it was not worth the mess it made.

raffclar commented on tgui/packages/tgui/routes.js:
```diff
@@ -57,14 +56,7 @@ export const getRoutedComponent = store => {
-    try {
-      esModule = requireInterface(interfacePath);
-    }
-    catch (err) {
```
This is the critical bit that loads the component for the player to see and interact with. I'm not sure what you're suggesting exactly but we need two try/catch attempts, one for fulp interfaces and the fallback for the original tg interfaces.

JohnFulpWillard commented on tgui/packages/tgui/routes.js:
```diff
@@ -57,14 +56,7 @@ export const getRoutedComponent = store => {
-        throw err;
-      }
-    }
+    esModule = loadInterface(interfacePath);
```
This is what I mean, only changing the file we need to.
I did this locally and it seems to work:
![image](https://user-images.githubusercontent.com/53777086/164590951-e4731d8e-44ff-4560-8706-ea79b75e2091.png)

It will make it easier to do future TGU's, especially since we barely know TGUI

```suggestion
    try {
      esModule = loadInterface(interfacePath);
    }
    catch (err) {
      if (err.code !== 'MODULE_NOT_FOUND') {
        throw err;
      }
    }
  }
```

raffclar replied:
I see what you mean now. I can do that sure

raffclar replied:
Sorted

JohnFulpWillard commented:
Last thing I want to bother you about, we have to distinguish these as TG edits, comments saying ``// Fulp edit - Fulp TGUI``, and adding them to tg_edits.md for TG updates in the future. Wanted to say it last so we can know what will be needed in the end or not.

raffclar commented:
I struggled to write tests for this so I'm dropping tests for now. There's no component-level tests upstream and there's something missing config settings. As Fulp tries to minimise merge conflicts it would be a bad idea to try add better test support.

@JohnFulpWillard
* Removed powers tab, removal is kept in a new commit.
* Took your advice and maintained the try/catch blocks.
* Added entries to the tg_edit doc and a few comments where possible.
* Fixed failing CI
